pkg/cli/haproxy_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2017 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cli
+
+import (
+	"testing"
+
+	"reflect"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/server/status"
+	"github.com/cockroachdb/cockroach/pkg/util"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+)
+
+func TestNodeStatusToNodeInfoConversion(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	input := []status.NodeStatus{
+		{
+			Desc: roachpb.NodeDescriptor{
+				NodeID: 1,
+				Address: util.UnresolvedAddr{
+					AddressField: "addr",
+				},
+			},
+			// Flags but no http port.
+			Args: []string{"--unwanted", "-unwanted"},
+		},
+		{
+			Args: []string{"--unwanted", "-http-port=1234"},
+		},
+		{
+			Args: []string{"--http-port", "5678", "--unwanted"},
+		},
+		{
+			// No flags.
+			Args: nil,
+		},
+		{
+			// We shouldn't see this, because the flag needs an argument on startup,
+			// but check that we fall back to the default port.
+			Args: []string{"-http-port"},
+		},
+	}
+	expected := []haProxyNodeInfo{
```

Comment:
Nit, but the convention throughout most of the code base is to include the expected results next to the input that generates them, rather than in separate slices that readers have to mentally join together. You can leave this is as if you want, though.

Reply:
No problem! Updated.